Code/GraphMol/MonomerMol/MonomerMol.h

```diff
@@ -0,0 +1,127 @@
+/* -------------------------------------------------------------------------
+ * Declares tools for Monomeristic ROMols
+ *
+ * A "MonomerMol" is an ROMol that uses RDKit atoms to represent monomers. Chains
+ * are represented via the PDBAtomResidueInfo structs on atoms, and linkages are
+ * stord as a LINKAGE property on bonds in the form of RX-RY, where X is the attachment
+ * point used on the begin monomer and Y is the attachment point used on the monomer.
+ *
+ *
+ * Copyright Schrodinger LLC, All Rights Reserved.
+ --------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <RDGeneral/export.h>
+
+namespace RDKit
+{
+
+// Forward declarations
+class ROMol;
+class RWMol;
+class Atom;
+class Bond;
+class SubstanceGroup;
+
+const std::string ANNOTATION{"ANNOTATION"};
+const std::string LINKAGE{"attachmentPoints"};
+const std::string ATOM_LABEL{"atomLabel"};
+const std::string SUPPLEMENTARY_INFORMATION{"SUPPLEMENTARY_INFORMATION"};
+const std::string BRANCH_LINKAGE{"R3-R1"};
+const std::string BACKBONE_LINKAGE{"R2-R1"};
+const std::string HELM_MODEL{"HELM_MODEL"};
+const std::string MONOMER_LIST{"MONOMER_LIST"};
+const std::string UNKNOWN_MONOMER{"UNKNOWN_MONOMER"};
+const std::string REPETITION_DUMMY_ID{"REPETITION_DUMMY_ID"};
+
+// NOTE: These are to allow replacement of the python api
+const std::string BRANCH_MONOMER{"isBranchMonomer"};
+const std::string SMILES_MONOMER{"isSmilesMonomer"};
+const std::string CUSTOM_BOND{"customBond"};
+
+// acts as a view to a polymer chain
+struct Chain {
+    std::vector<unsigned int> atoms;
+    std::vector<unsigned int> bonds;
+    std::string annotation;
+    // std::string polymer_id;
+};
+
+enum class ChainType { PEPTIDE, RNA, DNA, CHEM };
```

Comment:
Maybe these enums could be `BETTER_ENUM`s so that you get the enum<->string conversion automatically?

There are examples of these being used in `RGroupDecompParams`, `CXSmilesFields`, etc.